src/MBC_UserImport_Source_Niche.php

```diff
@@ -109,140 +120,70 @@ public function canProcess($message)
     }
 
-    if (isset($message['name']) && !isset($message['first_name'])) {
-      $nameBits = $this->mbcUserImportToolbox->nameBits($message['name']);
-      $this->importUser['first_name'] = $nameBits['first_name'];
-      $this->importUser['last_name'] = $nameBits['last_name'];
+    // Mobile.
+    if (!empty($message['phone'])) {
+      // Validate phone number based on the North American Numbering Plan
+      // https://en.wikipedia.org/wiki/North_American_Numbering_Plan
+      $pattern = "/^(\d[\s-]?)?[\(\[\s-]{0,2}?\d{3}[\)\]\s-]{0,2}?\d{3}[\s-]?\d{4}$/i";
+      if (preg_match($pattern, $message['phone']) !== false) {
+        $this->user['mobile'] = $message['phone'];
+      }
     }
 
-    $firstName = isset($message['first_name']) && $message['first_name'] != ''
-      ? $message['first_name'] : 'DS';
-    $this->importUser['password']
-      = str_replace(' ', '', $firstName) . '-Doer' . rand(1, 1000);
-
-    // Optional fields
-    if ($message['birthdate'] > time()) {
-      echo '- WARNING: Invalid birthdate: ' . $message['birthdate'] .
-        ' -> date(): ' . date('r', $message['birthdate']), PHP_EOL;
-    } else {
-      if (isset($message['birthdate']) && is_int($message['birthdate'])) {
-        $this->importUser['birthdate_timestamp'] = $message['birthdate'];
-      } elseif (isset($message['birthdate'])
-        && ctype_digit($message['birthdate'])
-      ) {
-        $this->importUser['birthdate_timestamp']
-          = (int) $message['birthdate'];
-      } elseif (isset($message['birthdate'])
-        && is_string($message['birthdate'])
-      ) {
-        $this->importUser['birthdate_timestamp']
-          = strtotime($message['birthdate']);
+    // Birthday.
+    if (!empty($message['birthday'])) {
+      if (is_int($message['birthdate']) || ctype_digit($message['birthdate'])) {
+        $this->user['birthdate'] = (int) $message['birthdate'];
+      } else {
+        $this->user['birthdate'] = strtotime($message['birthdate']);
       }
     }
+
+    // First name.
     if (!empty($message['first_name'])) {
-      $this->importUser['first_name'] = $message['first_name'];
-    }
-    if (!empty($message['first_name'])) {
-      $this->importUser['merge_vars']['FNAME'] = $message['first_name'];
-    }
-    if (!empty($message['last_name'])) {
-      $this->importUser['last_name'] = $message['last_name'];
+      $this->user['first_name'] = $message['first_name'];
     }
+
+    // Last name.
     if (!empty($message['last_name'])) {
-      $this->importUser['merge_vars']['LNAME'] = $message['last_name'];
-    }
-    if (!empty($message['password'])) {
-      $this->importUser['merge_vars']['PASSWORD'] = $message['password'];
+      $this->user['last_name'] = $message['last_name'];
     }
+
+    // Address.
     if (!empty($message['address1'])) {
-      $this->importUser['address1'] = $message['address1'];
+      $this->user['addr_street1'] = $message['address1'];
```

Comment:
Yes, good idea. Usually, I do it like this, just too used to legacy Quicksilver stile.